Python/pywarpx/picmi.py

```diff
@@ -61,9 +61,9 @@ def init(self, kw):
                     self.charge = '-q_e'
                 else:
                     self.charge = self.charge_state*constants.q_e
-            # Match a string of the format '#nXx', with the '#n' optional isotope number.
-            m = re.match(r'(?P<iso>#[\d+])*(?P<sym>[A-Za-z]+)', self.particle_type)
-            if m is not None:
+            if self.particle_type is not None:
+                # Match a string of the format '#nXx', with the '#n' optional isotope number.
+                m = re.match(r'(?P<iso>#[\d+])*(?P<sym>[A-Za-z]+)', self.particle_type)
                 element = periodictable.elements.symbol(m['sym'])
```

Comment:
Sorry, this block of code, from lines 67 to 80 (new line numbers) still needs to be inside the if test `if m is not None:` (indented one level more).
Also, something that was missing from the code is an error message when the `particle_type` is bad. This could be added here, adding an `else` clause that has the line
```
raise Exception('The species "particle_type" is not known')
```

Reply:
Thank you, Dave. I reinstated the `if m is not None:` condition in the last commit (force-pushed), hope I got it right this time.

Regarding the exception, I'm not sure we need it. I see that when I type in an unknown `particle_type` I get anyways the error `ValueError: unknown element blabla` after the call `element = periodictable.elements.symbol(m['sym'])`. The `else` clause that you suggested, where did you want it to be exactly (like, corresponding to which `if` clause)?